modules/core/site.js

```diff
@@ -2588,6 +2587,379 @@ function setupActionSnooze(callback) {
+        {
+          name: "filter_test_type",
+          value: $(".modal_sieve_filter_test").val(),
+        },
+        { name: "gen_script", value: gen_script },
+      ],
+      function (res) {
+        if (Object.keys(res.script_details).length === 0) {
+           window.location.href = "?page=sieve_filters";
+          
+        } else {
+          edit_script_modal.open();
+          $(".modal_sieve_script_textarea").val(res.script_details.gen_script);
+          $(".modal_sieve_script_name").val(res.script_details.filter_name);
+          $(".modal_sieve_script_priority").val(
+            res.script_details.filter_priority
+          );
+        }
+      }
+    );
+
+        return true;
+    };
+
+    my.save_script = function (imap_account) {
+        if ($('.modal_sieve_script_name').val() === "") {
+            Hm_Notices.show('You must provide a name for your script', 'warning');
+            return false;
+        }
+        if ($('.modal_sieve_script_textarea').val() === "") {
+            Hm_Notices.show('Empty script', 'warning');
+            return false;
+        }
+        Hm_Ajax.request(
+            [   {'name': 'hm_ajax_hook', 'value': 'ajax_sieve_save_script'},
+                {'name': 'imap_account', 'value': imap_account},
+                {'name': 'sieve_script_name', 'value': $('.modal_sieve_script_name').val()},
+                {'name': 'sieve_script_priority', 'value': $('.modal_sieve_script_priority').val()},
+                {'name': 'is_editing_script', 'value': is_editing_script},
+                {'name': 'current_editing_script', 'value': current_editing_script_name},
+                {'name': 'script', 'value': $('.modal_sieve_script_textarea').val()}],
+            function(res) {
+                window.location = window.location;
+            }
+        );
+    }
+
+    my.add_filter_condition = function() {
+      let header_fields = "";
+      let message_fields = "";
+
+      hm_sieve_condition_fields().Message.forEach(function (value) {
+        if (value.selected === true) {
+          message_fields +=
+            '<option selected value="' +
+            value.name +
+            '">' +
+            value.description +
+            "</option>";
+        } else {
+          message_fields +=
+            '<option value="' +
+            value.name +
+            '">' +
+            value.description +
+            "</option>";
+        }
+      });
+
+      hm_sieve_condition_fields().Header.forEach(function (value) {
+        if (value.selected === true) {
+          header_fields +=
+            '<option selected value="' +
+            value.name +
+            '">' +
+            value.description +
+            "</option>";
+        } else {
+          header_fields +=
+            '<option value="' +
+            value.name +
+            '">' +
+            value.description +
+            "</option>";
+        }
+      });
+
+      let extra_options =
```

Comment:
If we are building all this html inline, can you at least use template literals and spare the quotes on each line?